src/tools.rs

```diff
@@ -1,22 +1,25 @@
 //! Download management for external tools and applications. Locate and automatically download
```

Comment:
Sorry that you had to revert back to the blocking libs for this! The core issue was around async zip support?

Reply:
Yes, there is no async version of the `zip` crate and the API of that crate is not really good so even in sync code I had quite a hard time to get it working properly.

From the async perspective it's not really a loss going back to the normal `tar` crate because file based async IO is run in a thread pool anyways as there are no async file APIs on the OS level yet (there is io-uring for Linux but that still takes a while to be ready and widespread enough).